redash/models/__init__.py

```diff
@@ -354,7 +354,10 @@ def should_schedule_next(previous_iteration, now, interval, time=None, day_of_we
         next_iteration = (previous_iteration + datetime.timedelta(days=days_delay) +
                           datetime.timedelta(days=days_to_add)).replace(hour=hour, minute=minute)
     if failures:
-        next_iteration += datetime.timedelta(minutes=2**failures)
+        try:
```

Comment:
What you're saying is basically equivalent to "after failure 31, the next scheduled time is soooo far ahead in the future that there's really no point in scheduling it, meaning that the `should_schedule_next` function should simply return `False`, instead of a magic date in the post apocalyptic world. 😄 

How about this?

```suggestion
        try:
            next_iteration += datetime.timedelta(minutes=2**failures)
            return now > next_iteration
        except OverflowError:
            return False            
```

Reply:
Thanks. Make sense

Reply:
Seems like the suggestion wasn't applied correctly. Not sure Github is supporting multi-line suggestions ATM.

![image](https://user-images.githubusercontent.com/289488/59340620-033f0b80-8d0f-11e9-965e-56713db1b489.png)

Reply:
`return now > next_iteration` should be outside the `if failures` to cover also the non failure cases.
I fixed it accordingly